c++/src/capnp/capability.h

```diff
@@ -205,6 +205,17 @@ class Capability::Client {
   // Make a client capability that wraps the given server capability.  The server's methods will
   // only be executed in the given EventLoop, regardless of what thread calls the client's methods.
 
+  template <typename T, typename = kj::EnableIf<kj::canConvert<T*, Capability::Server*>()>>
+  Client(kj::Rc<T>&& server);
```

Comment:
It's actually sort of unsafe to have a capnp `Server` object be refcounted. In particular it means that `thisCap()` cannot be implemented correctly. I don't think we should encourage it.

Generally the right pattern is to always allocate server objects with `kj::heap`, and then, if you need to be able to refcount it, pass around the `Client` object and make clones of that.